src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -481,6 +477,7 @@ static void pf_connectivity_changed_locked(void* arg, grpc_error* error) {
       grpc_lb_subchannel_data_start_connectivity_watch(sd);
       break;
     }
+    case GRPC_CHANNEL_SHUTDOWN:
```

Comment:
If the subchannel that's not selected is in `SHUTDOWN`, we can still treat it as a `TRANSIENT_FAILURE` because the subchannel can retry. It doesn't hurt to keep watching it if we haven't selected a successful subchannel.

So I think maybe it's more accurate to keep the logic prior to this PR: the subchannel never shuts down.